### PR TITLE
Enhancement: [Image Cropping] Use new API from `canhub` 

### PR DESCRIPTION
**Pull request by criticalAY**
## Purpose / Description
Use the new API to handle and remove the deprecated API from canhub i.e. create our own activity 

Blocking on: 
* https://github.com/ankidroid/Anki-Android/pull/16798

## Fixes
* https://github.com/ankidroid/Anki-Android/pull/16824

## How Has This Been Tested?
Emulator API 35

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by criticalAY** [comment]
> w: file:///Users/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/test/java/com/ichi2/anki/ActivityStartupUnderBackupTest.kt:20:8 '@Deprecated(...) @SourceDebugExtension(...) class CropImageActivity : AppCompatActivity, CropImageView.OnSetImageUriCompleteListener, CropImageView.OnCropImageCompleteListener' is deprecated. Create your own Activity and use the CropImageView directly.
Will remove it when the blocker gets merged

**Comment by david-allison** [comment]
Note: you don't want a failing commit in the history, so thids should either be reordered or squashed

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt [hunk 1]
```diff
@@ -778,15 +781,33 @@ class NoteEditor : AnkiFragment(R.layout.note_editor), DeckSelectionListener, Su
...
+
+                else -> {
+                    Timber.v("Unable to crop the image")
+                }
             }
         }
+
+    private fun startCrop(imageUri: Uri) {
+        val intent = ImageCropperLauncher.ImageUri(imageUri).getIntent(requireContext())
+        cropImageLauncher.launch(intent)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:810]: Is there a reason there's so much code churn here?

File: AnkiDroid/src/main/java/com/ichi2/anki/multimedia/MultimediaImageFragment.kt [hunk 2]
```diff
@@ -167,6 +169,36 @@ class MultimediaImageFragment : MultimediaFragment(R.layout.fragment_multimedia_
             }
         }
 
+    /** Launches an activity to crop the image, using the [ImageCropper] */
+    private val imageCropperLauncher =
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimedia/MultimediaImageFragment.kt:173]: ditto

File: AnkiDroid/src/main/res/layout/fragment_image_cropper.xml [hunk 3]
```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="utf-8"?><!--
+  ~ Copyright (c) 2024 Ashish Yadav <mailtoashish693@gmail.com>
```
Comment (david-allison) [inline AnkiDroid/src/main/res/layout/fragment_image_cropper.xml:2]: is this correct?

File: AnkiDroid/src/main/java/com/ichi2/imagecropper/ImageCropper.kt [hunk 4]
```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2024 Ashish Yadav <mailtoashish693@gmail.com>
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/imagecropper/ImageCropper.kt:2]: Is this correct?

**Review by david-allison — COMMENTED** [review]
1. If we're copying their code in, I would expect a couple of lines change in our calling code
2. Copyright seems like we don't acknowledge our source.

See:

https://github.com/ankidroid/Anki-Android/blob/864b7c75b4725dfd7710d62a38e0d62d13b8dab9/AnkiDroid/src/main/java/com/ichi2/libanki/Sound.kt#L17-L24

File: AnkiDroid/src/main/res/layout/fragment_image_cropper.xml (hunk 3 above)
Comment (criticalAY) [reply AnkiDroid/src/main/res/layout/fragment_image_cropper.xml:2]: Yes, we have our own implementation, that is different from canhub. didn't copy this

File: AnkiDroid/src/main/java/com/ichi2/imagecropper/ImageCropper.kt (hunk 4 above)
Comment (criticalAY) [reply AnkiDroid/src/main/java/com/ichi2/imagecropper/ImageCropper.kt:2]: Majority of the code is customized according to our usage, we own it ?

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/imagecropper/ImageCropper.kt:2]: Give attribution to where you got it from

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt (hunk 1 above)
Comment (criticalAY) [reply AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.kt:810]: Earlier we did not had the specific activity for it

**Comment by mikehardy** [comment]
I conflicted this with all my dependency updates, so I took the penalty and de-conflicted it then re-pushed.
Also I needed it to re-run CI with new set of checks

**Comment by david-allison** [comment]
@criticalAY I'm self-assigning with the expectation that I'll get through this today. Please ping if this doesn't happen

I still feel there's too much code churn here, and I'm a little concerned about the copyright notices, so I want to work through this

Please ping if I'm not done within 24 hours

**Comment by david-allison** [comment]
I'm posting PR feedback which I would provide here. I'm going to edit this post so it won't spam notifications

This is non-actionable and informative only, as I'll resolve these:

* The commit message needs much more information about where the code came from and why it occurred

**Comment by david-allison** [comment]
@criticalAY After a quick look, I'm asking for busywork

Please improve the commit message significantly, explaining the choices you made. Here's a start:

> refactor(crop-image): fix `CropImageContract` deprecation
> 
> `CropImageContract` will be deprecated in `android-image-cropper:4.6.0`
as the library would like to focus on only providing cropping functionality
>
> Their defaults also included other unrelated functionality: loading
images etc..., and this is being moved into consumers of the library
> 
> The recommended upgrade path is to use `CropImageView`
> 
> * https://redirect.github.com/CanHub/Android-Image-Cropper/pull/637

**Review by david-allison — CHANGES_REQUESTED** [review]
> "feat: use CropImageView to **crop crop** images"

**Comment by mikehardy** [comment]
I would like to see this go in, so I fixed the conflict (simple import statement thing) which simultaneously allowed me to reduce the `crop crop` in the comment to `crop`, so hopefully this is ready for serious review now

**Review by david-allison — APPROVED** [review]
I was previously an 'approve', and am still

With the note that the build now fails and that will need to be resolved, code structure is fine

**Review by mikehardy — APPROVED** [review]
Alright then, let's go! Thanks @criticalAY

**Pull request merged**